scripts/content/VACMS-6931-switch-service-locations-office-hours.php

```diff
@@ -0,0 +1,378 @@
+      $hours_clean[$low_day]['id'] = $service_location->id();
+      $hours_clean[$low_day]['raw'] = $raw_value;
+
+      // Normalize meridiem designations.
+      $new_value = normalize_meridiem($raw_value);
+
+      // Normalize time separator designations.
+      $new_value = normalize_separator($new_value);
+
+      // Gather some initial data about our new string.
+      preg_match_all('/(a\.m\.|p\.m\.)/', $new_value, $meridiems);
+      $meridiem_count = count($meridiems[0]);
+      preg_match_all('/(ZZZ)/', $new_value, $separators);
+      $separator_count = count($separators[0]);
+
+      // If the string doesn't start with a number, treat as a comment.
+      if (!ctype_digit(substr(trim($raw_value), 0, 1))) {
+        $hours_clean[$low_day]['comment'] = comment_cleaner($raw_value);
+      }
+      elseif ($meridiem_count != 2) {
+        // If the number of meridiems is not two, treat as a comment.
+        $hours_clean[$low_day]['comment'] = comment_cleaner($raw_value);
+      }
+      elseif (!$separator_count || $separator_count > 2) {
+        // If the separator count is neither 1 nor 2, treat as a comment.
+        $hours_clean[$low_day]['comment'] = comment_cleaner($raw_value);
+      }
+      else {
+        // Break up the normalized string by the separator.
+        $pieces = explode('ZZZ', $new_value);
+
+        // The first two pieces must contain meridiems to be treated as a time.
+        if ((preg_match('/(a\.m\.|p\.m\.)/', $pieces[0]) === 0)
+        || (preg_match('/(a\.m\.|p\.m\.)/', $pieces[1]) === 0)) {
+          // If no meridiems in first two pieces, treat as a comment.
+          $hours_clean[$low_day]['comment'] = comment_cleaner($raw_value);
+        }
+        else {
+          // Process the first piece as a time value.
+          $start_time = clean_time($pieces[0]);
+          $sub_pieces = [];
+
+          // The second piece could still contain a comment.
+          if (str_contains($pieces[1], 'a.m.')) {
+            $sub_pieces = explode('a.m.', $pieces[1]);
+            $end_time = clean_time($sub_pieces[0] . 'a.m.');
+          }
+          elseif (str_contains($pieces[1], 'p.m.')) {
+            $sub_pieces = explode('p.m.', $pieces[1]);
+            $end_time = clean_time($sub_pieces[0] . 'p.m.');
+          }
+
+          // If the start time is greater than the end time, treat as a comment.
+          if ($start_time > $end_time) {
+            $hours_clean[$low_day]['comment'] = comment_cleaner($raw_value);
+          }
+          else {
+            // Set the time values.
+            $hours_clean[$low_day]['start_time'] = $start_time;
+            $hours_clean[$low_day]['end_time'] = $end_time;
+
+            // Add any remaining sub_pieces to comment.
+            array_shift($sub_pieces);
+            $comment = '';
+            if (count($sub_pieces)) {
+              foreach ($sub_pieces as $sub_piece) {
+                $comment .= $sub_piece;
+              }
+            }
```

Comment:
Would implode work here instead of foreach?
```suggestion
            $comment = implode(' ', $sub_pieces);    
```